src/app.rs

```diff
@@ -21,24 +21,106 @@
 //! - UDP trackers: the user can enable multiple UDP tracker on several ports.
 //! - HTTP trackers: the user can enable multiple HTTP tracker on several ports.
 //! - Tracker REST API: the tracker API can be enabled/disabled.
+use std::collections::HashMap;
+use std::net::SocketAddr;
 use std::sync::Arc;
 
+use derive_more::Constructor;
 use log::warn;
+use tokio::sync::Mutex;
 use tokio::task::JoinHandle;
 use torrust_tracker_configuration::Configuration;
 
 use crate::bootstrap::jobs::{health_check_api, http_tracker, torrent_cleanup, tracker_apis, udp_tracker};
 use crate::{core, servers};
 
+/// Returned by the service check.
+#[derive(Debug, Constructor)]
+pub struct CheckJob {
+    pub binding: SocketAddr,
+    pub info: String,
+    pub job: JoinHandle<Result<String, String>>,
+}
+
+/// The function type that is called to check a service.
+pub type FnCheck = fn(&SocketAddr) -> CheckJob;
+
+/// A [`Listing`] is provided to the [`Registar`] for registration.
+#[derive(Clone, Debug, Constructor)]
+pub struct Listing {
+    binding: SocketAddr,
+    check_fn: FnCheck,
+}
+
+/// A [`RegistrationForm`] is completed with a [`Listing`] and then is sent.
+pub type RegistrationForm = tokio::sync::oneshot::Sender<Listing>;
```

Comment:
hi @da2ce7 I'm a little bit concerned about the name `RegistrationForm`. I think it's used very often for signups. We could call it:

- `ServiceRegistrationForm`

By the way, the main tracker app runs different services:

- Tracker API
- HTTP tracker
- UDP tracker

I've been calling them services.

On the other hand, why did you use the name `Listing`?

![image](https://github.com/torrust/torrust-tracker/assets/58816/b5e2a075-c42d-4f97-a677-1cb04ef8ca62)

Maybe `RegistryEntry` would be more explicit. For me `listing` is more a list than an entry on a list, but maybe because I'm not a native English speaker.

Reply:
Done :smile: